add shape check for rrelu_with_noise

jiayisunx opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122870

Fix https://github.com/pytorch/pytorch/issues/121094.
Add shape check for rrelu_with_noise, check whether the shape of input tensor and noise tensor are the same.

jiayisunx committed:
add shape check for rrelu_with_noise

[ghstack-poisoned]

Skylion007 commented on aten/src/ATen/native/Activation.cpp:
```diff
@@ -606,6 +606,7 @@ Tensor& rrelu_with_noise_out_cpu(const Tensor& self,
     bool training,
     c10::optional<Generator> generator,
     Tensor& output) {
+  TORCH_CHECK(self.sizes() == noise.sizes(), "noise tensor shape must match self tensor shape. Got self.shape = ", self.sizes(), " noise.shape = ", noise.sizes())
```
Should this be `sym_sizes()` to prevent issues with symbolic tensors?

jiayisunx replied:
Fixed, thanks!

jiayisunx committed:
Update on "add shape check for rrelu_with_noise"


Fix https://github.com/pytorch/pytorch/issues/121094.
Add shape check for rrelu_with_noise, check whether the shape of input tensor and noise tensor are the same.

[ghstack-poisoned]

jiayisunx committed:
Update on "add shape check for rrelu_with_noise"


Fix https://github.com/pytorch/pytorch/issues/121094.
Add shape check for rrelu_with_noise, check whether the shape of input tensor and noise tensor are the same.

[ghstack-poisoned]

jiayisunx commented:
@Skylion007, could you please review this PR again? Thanks.

mingfeima commented on aten/src/ATen/native/Activation.cpp:
```diff
@@ -606,6 +606,7 @@ Tensor& rrelu_with_noise_out_cpu(const Tensor& self,
     bool training,
     c10::optional<Generator> generator,
     Tensor& output) {
+  TORCH_CHECK(self.sym_sizes() == noise.sym_sizes(), "noise tensor shape must match self tensor shape. Got self.shape = ", self.sym_sizes(), " noise.shape = ", noise.sym_sizes())
```
TORCH_CHECK does not need a `;` to be compiled correctly, but usually we still add a `;` at the end

jiayisunx committed:
Update on "add shape check for rrelu_with_noise"


Fix https://github.com/pytorch/pytorch/issues/121094.
Add shape check for rrelu_with_noise, check whether the shape of input tensor and noise tensor are the same.

[ghstack-poisoned]

jiayisunx commented:
@pytorchbot merge

jiayisunx commented:
@cpuhrsch  , could you please review this PR? Thanks.

ezyang commented:
@pytorchbot merge
